jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java

```diff
@@ -96,43 +103,91 @@ private void createConsumer() throws InterruptedException {
-                    LOGGER.error("HonoClient.createEventConsumer() failed", creationHandler.cause());
-                }
-                receiverLatch.countDown();
-            });
+            amqpNetworkClient.createEventConsumer(sampler.getTenant(), Integer.parseInt(sampler.getPrefetch()),
+                    this::messageReceived, creationHandler -> {
+                        if (creationHandler.failed()) {
+                            LOGGER.error("HonoClient.createEventConsumer() failed", creationHandler.cause());
+                        }
+                        receiverLatch.countDown();
+                    });
         }
         receiverLatch.await();
     }
 
-    public void sample(final SampleResult result, final boolean isUseSenderTime) {
+    public void sample(final SampleResult result) {
         synchronized (lock) {
-            long elapsed = System.currentTimeMillis() - sampleStart;
+            long elapsed = 0;
             result.setResponseCodeOK();
             result.setSuccessful(true);
             result.setSampleCount(messageCount);
-            result.setResponseMessage(MessageFormat.format("count: {0}, bytes received: {1}, period: {2}", messageCount, messageSize, elapsed));
             result.setBytes(messageSize);
-            if (messageCount > 0) {
+            result.setResponseMessage(
+                    MessageFormat.format("count: {0}, bytes received: {1}, period: {2}", messageCount,
+                            messageSize, elapsed));
+            if (sampler.isUseSenderTime()) {
+                if (messageCount > 1) {
+                    LOGGER.warn(
+                            "More than one message per receiver was not expected. It could result in wrong sampling time");
+                } else if (messageCount == 1) {
+                    elapsed = sampleEnd - sampleStart;
+                    result.setStampAndTime(sampleStart, elapsed);
+                } else {
+                    noMessagesReceived(result);
+                }
+            } else if (messageCount > 0 && sampleStart != 0) {
+                elapsed = System.currentTimeMillis() - sampleStart;
                 result.setStampAndTime(sampleStart, elapsed);
+                result.setIdleTime(elapsed);
             } else {
-                result.setStampAndTime(sampleStart, 0);
+                noMessagesReceived(result);
             }
-            LOGGER.info("{}: received batch of {} messages in {} milliseconds", sampler.getThreadName(), messageCount, elapsed);
+            LOGGER.info("{}: received batch of {} messages in {} milliseconds", sampler.getThreadName(), messageCount,
+                    elapsed);
             messageSize = 0;
             messageCount = 0;
-            sampleStart = System.currentTimeMillis();
+            sampleStart = 0;
+        }
+    }
+
+    private void noMessagesReceived(final SampleResult result) {
+        LOGGER.warn("No messages were received");
+        result.setIdleTime(0);
```

Comment:
How is this used by JMeter in the results? Does idle time 0 means no idle time? But why is this, if there have been not messages between to samples?

Reply:
So when we start more than one receiver (in our case we start the publisher/receiver in a loop), sometimes a receiver doesn't receive any message as it might have started late.

Reply:
I wondered, because before the time was explicitly set to 0 if no messages has been received to archive this. And I was not sure, that the call of setIdleTime(0) is doing the same. (Because the documentation of the JMeter API is .. challenging) ;) If you sure, I am fine with this.

Reply:
As we are taking time stamp from the message received, with setIdleTime(0), I wanted to make that explicit even if the receiver was waiting for sometime to receive messages.